### DISCUSSION TITLE
3D Scene Extension with Interaction

### DISCUSSION THREAD

**Discussion by mechatheo** [foxglove/community#895] [Q&A]
Hello Foxglove Team,

we are thinking of switching to Foxglove (from our good old Rviz-based toolchain). A tool that we always had in Rviz is Draw Path. So you could draw on a scene and then it would publish the drawn path as a Nav/Path in the currently selected frame to a message of your choice (or even call an action like exe_path).

I looked into the foxglove extension examples but for me, it became not quite clear whether it's easy to do the same kind of tool with foxglove. I saw that custom panels can be added and message conversion plugins are available. But I would need to hook into the existing 3D view, is something like this even possible as of right now? And if so, could someone provide some pointers on where you would start, even if workarounds are required?

This is roughly what I would like to have:
![image](https://github.com/foxglove/community/assets/20418799/933788a9-9b09-436a-ba32-64a48816908b)

Kind Regards
Theo

**Comment by defunctzombie** [comment]
> But I would need to hook into the existing 3D view, is something like this even possible as of right now?

This is not currently possible with the extension interface. Either we'd need to expose various ways to add custom logic like this to the 3d panel or alternatively you'd be looking at introducing this feature in upstream 3d panel code.
